### PR TITLE
add size to album details

### PR DESCRIPTION
**Pull request by jrwren**
for #534

### CODE REVIEW

**Comment by jrwren** [comment]
The album details part looks like this now:

image

File: Makefile [hunk 1]
```diff
@@ -39,7 +39,7 @@ update-snapshots: check_go_env
 
 migration:
 	@if [ -z "${name}" ]; then echo "Usage: make migration name=name_of_migration_file"; exit 1; fi
-	goose -dir db/migrations create ${name}
+	goose -dir db/migration create ${name}
```
Comment (deluan) [inline Makefile:42]: Ops, thanks for this :)

File: db/migration/20201010162350_add_album_size.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	goose.AddMigration(Up20201010162350, Down20201010162350)
+}
+
+func Up20201010162350(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+alter table album
+	add size integer default 0 not null;
+	`)
+	return err
+	/*  IS A RESCAN NECESSARY TO UPDATE ALBUM TABLE WITH SIZES?
```
Comment (deluan) [inline db/migration/20201010162350_add_album_size.go:19]: Yes, the rescan will be necessary, as the sizes will be collected during the scan process

File: db/migration/20201010162350_add_album_size.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+func init() {
+	goose.AddMigration(Up20201010162350, Down20201010162350)
+}
+
+func Up20201010162350(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+alter table album
+	add size integer default 0 not null;
```
Comment (deluan) [inline db/migration/20201010162350_add_album_size.go:16]: Maybe also add an index for `size`, so in the future we can implement a "sort by size" in the UI?

File: ui/src/album/AlbumDetails.js [hunk 4]
```diff
@@ -49,6 +49,7 @@ const AlbumDetails = ({ classes, record }) => {
           {record.songCount}{' '}
           {translate('resources.song.name', { smart_count: record.songCount })}{' '}
           · <DurationField record={record} source={'duration'} />
+          · <SizeField record={record} source="size" />
```
Comment (deluan) [inline ui/src/album/AlbumDetails.js:52]: You should add a `{' '}` between the two fields, as you can see from your screenshot that the `·` is too close to the Duration time

File: persistence/album_repository.go [hunk 5]
```diff
@@ -156,6 +156,7 @@ func (r *albumRepository) refresh(ids ...string) error {
 		Years         string
 		DiscSubtitles string
 		Path          string
+		Size          int
```
Comment (deluan) [inline persistence/album_repository.go:159]: This line is not required and can be removed. This `refreshAlbum` struct is for extra columns that are needed for post-processing the query results, if the value for a field in the result set is one-to-one to the expected album value, you don't need to do anything else.

**Review by deluan — CHANGES_REQUESTED** [review]
Great job! Just some minor adjustments before merging.

**Comment by deluan** [comment]
It would also be great to add `size` to the album details, so they are available in the Album list view: https://github.com/deluan/navidrome/blob/e6d1e67297295ab061575e449872fc00cc5aab2d/ui/src/album/AlbumListView.js#L32-L43

**Comment by deluan** [comment]
LGTM. Thanks!

**Comment by jrwren** [comment]
any chance in getting a hacktoberfest-accepted label on this PR?

**Comment by deluan** [comment]
Sure, I'll add the [topic to the project](https://hacktoberfest.digitalocean.com/hacktoberfest-update) and the label to the PR

**Pull request merged**